In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

%matplotlib inline
plt.style.use('default')
plt.rcParams['figure.figsize'] = (22, 12)
sns.set(style="whitegrid")
import warnings
warnings.filterwarnings('ignore')

# Events

In [3]:
events = pd.read_csv('./Datos2019/events.csv', low_memory=False)
events.head()

date  event_id             ref_type  \
0  2019-03-05 00:09:36.966         0  1891515180541284343   
1  2019-03-05 00:09:38.920         1  1891515180541284343   
2  2019-03-05 00:09:26.195         0  1891515180541284343   
3  2019-03-05 00:09:31.107         2  1891515180541284343   
4  2019-03-09 21:00:36.585         3  1891515180541284343   

              ref_hash  application_id  attributed   device_countrycode  \
0  2688759737656491380              38       False  6333597102633388268   
1  2688759737656491380              38       False  6333597102633388268   
2  2688759737656491380              38       False  6333597102633388268   
3  2688759737656491380              38       False  6333597102633388268   
4  2635154697734164782              38       False  6333597102633388268   

   device_os_version  device_brand  device_model       ...        trans_id  \
0       5.908703e+17           NaN  5.990117e+18       ...             NaN   
1       5.908703e+17           NaN  5.990117e+18       ...             NaN   
2       5.908703e+17           NaN  5.990117e+18       ...             NaN   
3       5.908703e+17           NaN  5.990117e+18       ...             NaN   
4       7.391844e+18           NaN  5.960896e+18       ...             NaN   

   user_agent                            event_uuid       carrier  \
0         NaN  a9c0b263-acb2-4577-92c5-cbde5d7a5db1  2.248157e+17   
1         NaN  1cd98205-0d97-4ec2-a019-667997dbfe7a  2.248157e+17   
2         NaN  f02e2924-21ae-492b-b625-9021ae0a4eca  2.248157e+17   
3         NaN  a813cf45-a36e-4668-85e2-5395f1564e98  2.248157e+17   
4         NaN  63a4f0aa-e147-469f-8c55-4ca4f8d0e310  2.248157e+17   

           kind     device_os  wifi  connection_type           ip_address  \
0  5.516623e+18  7.531669e+18   NaN        Cable/DSL  7858558567428669000   
1  9.977660e+17  7.531669e+18   NaN        Cable/DSL  7858558567428669000   
2  5.516623e+18  7.531669e+18   NaN        Cable/DSL  7858558567428669000   
3  8.561153e+18  7.531669e+18   NaN        Cable/DSL  6324037615828123965   
4  8.731902e+17  7.531669e+18   NaN        Cable/DSL  2894495631302821483   

  device_language  
0    4.077062e+17  
1    4.077062e+17  
2    4.077062e+17  
3    4.077062e+17  
4    3.301378e+18  

[5 rows x 22 columns]

In [3]:
events.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2494423 entries, 0 to 2494422
Data columns (total 22 columns):
date                  object
event_id              int64
ref_type              int64
ref_hash              int64
application_id        int64
attributed            bool
device_countrycode    int64
device_os_version     float64
device_brand          float64
device_model          float64
device_city           float64
session_user_agent    float64
trans_id              object
user_agent            float64
event_uuid            object
carrier               float64
kind                  float64
device_os             float64
wifi                  object
connection_type       object
ip_address            int64
device_language       float64
dtypes: bool(1), float64(10), int64(6), object(5)
memory usage: 402.0+ MB


In [4]:
events.describe()

event_id      ref_type      ref_hash  application_id  \
count  2.494423e+06  2.494423e+06  2.494423e+06    2.494423e+06   
mean   9.970445e+01  1.794164e+18  4.641486e+18    9.910934e+01   
std    1.070903e+02  1.707944e+17  2.660724e+18    5.780986e+01   
min    0.000000e+00  1.494519e+18  1.633675e+14    0.000000e+00   
25%    2.200000e+01  1.891515e+18  2.326142e+18    6.300000e+01   
50%    3.100000e+01  1.891515e+18  4.688964e+18    6.600000e+01   
75%    1.610000e+02  1.891515e+18  6.927594e+18    1.450000e+02   
max    7.120000e+02  1.891515e+18  9.223316e+18    3.220000e+02   

       device_countrycode  device_os_version  device_brand  device_model  \
count        2.494423e+06       1.022066e+06  1.164963e+06  2.406456e+06   
mean         6.333597e+18       4.986001e+18  1.633891e+18  4.478847e+18   
std          1.024000e+03       2.394834e+18  1.626674e+18  2.718014e+18   
min          6.333597e+18       1.004084e+16  7.949737e+16  9.530216e+14   
25%          6.333597e+18       4.353750e+18  3.083059e+17  2.331947e+18   
50%          6.333597e+18       4.353750e+18  5.137992e+17  4.430579e+18   
75%          6.333597e+18       7.391844e+18  2.987569e+18  6.882415e+18   
max          6.333597e+18       9.135482e+18  9.178260e+18  9.222450e+18   

        device_city  session_user_agent    user_agent       carrier  \
count  6.146980e+05        2.482637e+06  1.391527e+06  6.164340e+05   
mean   3.832628e+18        4.719312e+18  4.856492e+18  1.470186e+18   
std    8.166911e+17        1.769561e+18  2.487552e+18  2.575962e+18   
min    4.355428e+16        5.527631e+15  5.072532e+15  2.248157e+17   
25%    3.800080e+18        3.819516e+18  2.723465e+18  2.248157e+17   
50%    3.800080e+18        3.819516e+18  5.022936e+18  2.248157e+17   
75%    3.800080e+18        7.164321e+18  6.840860e+18  1.179587e+18   
max    9.205669e+18        9.220949e+18  9.223277e+18  9.182957e+18   

               kind     device_os    ip_address  device_language  
count  2.489324e+06  6.576670e+05  2.494423e+06     2.406604e+06  
mean   5.364362e+18  7.251101e+18  4.620786e+18     5.865447e+18  
std    2.242979e+18  5.247066e+17  2.672746e+18     2.281192e+18  
min    7.758270e+16  2.748831e+18  5.287755e+12     2.025809e+16  
25%    4.647949e+18  6.941825e+18  2.333410e+18     3.301378e+18  
50%    5.500848e+18  7.531669e+18  4.541498e+18     6.804428e+18  
75%    6.169667e+18  7.531669e+18  7.044307e+18     6.977049e+18  
max    9.215473e+18  7.531669e+18  9.223367e+18     9.187882e+18

In [5]:
events.device_countrycode.value_counts()

6333597102633388268    2494423
Name: device_countrycode, dtype: int64

# Installs

In [4]:
installs = pd.read_csv('./Datos2019/installs.csv', low_memory=False)
installs.head()

created  application_id             ref_type  \
0  2019-03-13 01:43:33.445               0  1891515180541284343   
1  2019-03-13 04:58:35.078               0  1891515180541284343   
2  2019-03-13 04:20:57.666               0  1891515180541284343   
3  2019-03-13 04:20:57.698               0  1891515180541284343   
4  2019-03-10 22:24:56.684               0  1891515180541284343   

              ref_hash  click_hash  attributed  implicit   device_countrycode  \
0  8464844987297247076         NaN       False     False  6333597102633388268   
1  3250564871270161533         NaN       False     False  6333597102633388268   
2  7953318831018100268         NaN       False      True  6333597102633388268   
3  7953318831018100268         NaN       False     False  6333597102633388268   
4  8355110941557237501         NaN       False      True  6333597102633388268   

   device_brand  device_model                 session_user_agent user_agent  \
0  3.083059e+17  4.445252e+18                                NaN        NaN   
1  3.083059e+17  4.445252e+18                                NaN        NaN   
2  5.137992e+17  6.026543e+18  HasOffers Mobile AppTracking v1.0        NaN   
3  5.137992e+17  6.026543e+18                                NaN        NaN   
4  1.083369e+18  7.876612e+18  HasOffers Mobile AppTracking v1.0        NaN   

                             event_uuid  kind wifi trans_id  \
0                                   NaN   NaN  NaN      NaN   
1                                   NaN   NaN  NaN      NaN   
2  0bfa9b4c-e07a-4522-b32d-138e3165ac98  Open  NaN      NaN   
3                                   NaN   NaN  NaN      NaN   
4  8f235a8c-41bf-4399-b33d-c41577c949b0  Open  NaN      NaN   

            ip_address  device_language  
0  5529284367937637301              NaN  
1  6098022721363533304              NaN  
2  4636407223847323867     3.301378e+18  
3  4636407223847323867              NaN  
4   685400811232710215     3.301378e+18

In [151]:
installs.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3412 entries, 0 to 3411
Data columns (total 18 columns):
created               3412 non-null object
application_id        3412 non-null int64
ref_type              3412 non-null int64
ref_hash              3412 non-null int64
click_hash            0 non-null float64
attributed            3412 non-null bool
implicit              3412 non-null bool
device_countrycode    3412 non-null int64
device_brand          1047 non-null float64
device_model          3411 non-null float64
session_user_agent    3364 non-null object
user_agent            1729 non-null object
event_uuid            865 non-null object
kind                  865 non-null object
wifi                  1729 non-null object
trans_id              6 non-null object
ip_address            3412 non-null int64
device_language       3378 non-null float64
dtypes: bool(2), float64(4), int64(5), object(7)
memory usage: 433.2+ KB


In [5]:
#Marcas
installs.device_brand.value_counts()

3.083059e+17    472
2.208835e+18    185
2.523246e+18    116
5.951325e+18     72
2.987569e+18     40
3.812621e+18     37
5.137992e+17     34
6.538562e+18     26
3.228516e+18      9
1.083369e+18      9
4.371308e+18      7
3.093166e+18      5
2.262848e+18      5
4.994101e+18      4
3.849490e+18      4
8.285870e+18      3
3.742387e+18      3
5.645692e+18      3
8.230045e+17      3
6.111647e+18      3
9.169527e+18      1
5.588755e+18      1
3.172231e+18      1
5.189983e+18      1
3.624089e+18      1
7.855186e+18      1
2.748831e+18      1
Name: device_brand, dtype: int64

In [6]:
#Modelo
installs.device_model.value_counts()

2.331947e+18    759
6.882415e+18    276
2.616228e+18     76
7.787531e+18     65
2.928262e+18     63
9.149036e+18     56
1.187661e+17     55
1.267677e+18     53
5.990117e+18     46
2.136935e+18     41
7.116343e+18     40
2.066431e+17     40
1.754528e+18     38
2.983638e+18     38
9.098443e+18     36
8.656854e+18     36
2.435450e+18     31
5.580205e+18     30
8.163783e+18     30
4.445252e+18     28
1.990627e+18     27
4.349816e+18     25
6.870743e+18     23
5.960896e+18     22
7.628197e+18     22
3.403532e+17     21
1.764701e+18     21
6.852389e+18     20
8.604122e+18     19
6.642035e+18     19
               ... 
4.835257e+18      1
3.272099e+18      1
7.723630e+18      1
3.320202e+18      1
4.469166e+18      1
1.464153e+18      1
6.897633e+17      1
7.252441e+16      1
7.276820e+18      1
2.863247e+18      1
5.777564e+18      1
5.680006e+18      1
2.577800e+18      1
8.531839e+18      1
8.399398e+18      1
8.304824e+18      1
4.059673e+18      1
4.008000e+18      1
4.073186e+18      1


In [152]:
installs.device_countrycode.value_counts()
#Hay dos codigos de pais

6333597102633388268    1820
2970470518450881158    1592
Name: device_countrycode, dtype: int64

In [10]:
installs.ref_type.value_counts()

1891515180541284343    2080
1494519392962156891    1332
Name: ref_type, dtype: int64

In [11]:
installs.device_language.value_counts()

3.301378e+18    1484
9.028383e+18     714
4.621024e+18     632
6.977049e+18     238
4.077062e+17      70
1.526421e+18      56
4.060930e+18      48
2.822843e+17      39
6.035180e+18      24
1.193279e+18      21
6.535228e+18      10
1.101539e+17       9
3.177265e+18       5
3.695787e+17       5
3.095856e+18       4
6.569908e+18       2
5.846217e+18       2
7.774316e+18       2
4.594650e+18       2
6.782349e+18       1
5.061642e+18       1
1.686906e+18       1
5.794427e+17       1
2.488366e+18       1
3.544544e+18       1
3.074904e+18       1
2.061937e+18       1
4.073835e+18       1
6.846873e+18       1
7.045130e+18       1
Name: device_language, dtype: int64

In [75]:
dispositivos = installs.loc[:,["created","device_countrycode", "device_brand","device_model","device_language"],]
dispositivos["Cantidad"] = 1
dispositivos

created   device_countrycode  device_brand  \
0     2019-03-13 01:43:33.445  6333597102633388268  3.083059e+17   
1     2019-03-13 04:58:35.078  6333597102633388268  3.083059e+17   
2     2019-03-13 04:20:57.666  6333597102633388268  5.137992e+17   
3     2019-03-13 04:20:57.698  6333597102633388268  5.137992e+17   
4     2019-03-10 22:24:56.684  6333597102633388268  1.083369e+18   
5     2019-03-10 22:24:57.545  6333597102633388268  1.083369e+18   
6     2019-03-11 03:47:22.802  6333597102633388268  3.083059e+17   
7     2019-03-11 21:21:05.986  6333597102633388268  1.083369e+18   
8     2019-03-11 21:21:05.855  6333597102633388268  1.083369e+18   
9     2019-03-13 08:29:31.875  6333597102633388268  3.083059e+17   
10    2019-03-13 04:30:00.812  6333597102633388268  3.083059e+17   
11    2019-03-13 00:26:32.561  6333597102633388268  2.208835e+18   
12    2019-03-13 11:36:30.025  6333597102633388268  2.987569e+18   
13    2019-03-13 13:15:14.566  6333597102633388268  3.083059e+17   
14    2019-03-13 16:52:25.368  6333597102633388268  2.208835e+18   
15    2019-03-13 15:16:35.506  6333597102633388268  2.523246e+18   
16    2019-03-13 15:02:31.785  6333597102633388268  3.083059e+17   
17    2019-03-13 19:17:49.840  6333597102633388268  2.523246e+18   
18    2019-03-13 19:17:55.554  6333597102633388268  2.523246e+18   
19    2019-03-13 19:32:49.002  6333597102633388268  2.208835e+18   
20    2019-03-13 18:42:57.790  6333597102633388268  2.208835e+18   
21    2019-03-13 01:38:59.934  6333597102633388268  3.083059e+17   
22    2019-03-13 01:41:52.323  6333597102633388268  3.083059e+17   
23    2019-03-13 01:41:31.932  6333597102633388268  3.083059e+17   
24    2019-03-13 23:48:03.589  6333597102633388268  2.523246e+18   
25    2019-03-13 23:09:55.098  6333597102633388268  2.523246e+18   
26    2019-03-09 23:28:46.033  6333597102633388268  2.208835e+18   
27    2019-03-09 00:16:33.738  6333597102633388268  3.083059e+17   
28    2019-03-09 00:48:36.248  6333597102633388268  3.083059e+17   
29    2019-03-09 22:14:33.927  6333597102633388268  3.083059e+17   
...                       ...                  ...           ...   
3382  2019-03-05 16:27:56.465  6333597102633388268  5.951325e+18   
3383  2019-03-05 20:54:16.969  6333597102633388268  5.951325e+18   
3384  2019-03-05 15:26:39.601  6333597102633388268  5.951325e+18   
3385  2019-03-05 13:38:13.806  6333597102633388268  5.951325e+18   
3386  2019-03-08 17:57:17.644  6333597102633388268  5.951325e+18   
3387  2019-03-08 08:47:31.667  6333597102633388268  5.951325e+18   
3388  2019-03-08 23:40:49.972  6333597102633388268  5.951325e+18   
3389  2019-03-07 04:21:29.011  6333597102633388268  3.083059e+17   
3390  2019-03-08 12:21:02.350  2970470518450881158           NaN   
3391  2019-03-08 19:02:51.398  2970470518450881158           NaN   
3392  2019-03-08 16:29:01.962  2970470518450881158           NaN   
3393  2019-03-08 22:20:08.795  2970470518450881158           NaN   
3394  2019-03-08 22:18:49.156  2970470518450881158           NaN   
3395  2019-03-07 17:10:59.636  2970470518450881158           NaN   
3396  2019-03-07 16:39:57.285  2970470518450881158           NaN   
3397  2019-03-07 02:04:20.316  2970470518450881158           NaN   
3398  2019-03-07 15:33:59.181  2970470518450881158           NaN   
3399  2019-03-07 14:32:12.349  2970470518450881158           NaN   
3400  2019-03-07 21:41:42.173  2970470518450881158           NaN   
3401  2019-03-07 22:59:31.466  2970470518450881158           NaN   
3402  2019-03-07 14:23:56.562  2970470518450881158           NaN   
3403  2019-03-05 00:13:57.727  2970470518450881158           NaN   
3404  2019-03-05 21:13:35.312  2970470518450881158           NaN   
3405  2019-03-05 21:45:59.896  2970470518450881158           NaN   
3406  2019-03-06 12:34:37.990  2970470518450881158           NaN   
3407  2019-03-06 17:49:41.192  2970470518450881158           NaN   
3408  2019-03-06 11:18:22.500  2970470518450881158           NaN   
3409  2019

In [76]:
#Veo cuantos devide_model hay en cada pais
dispositivos.groupby(["device_countrycode","device_model"]).agg({'Cantidad': 'sum'})

Cantidad
device_countrycode  device_model          
2970470518450881158 1.065211e+17         1
                    1.187661e+17        48
                    1.819881e+17         2
                    2.066431e+17         9
                    2.365619e+17         1
                    2.642581e+17        15
                    2.810099e+17        12
                    3.403532e+17        21
                    3.686067e+17         4
                    3.817244e+17         1
                    3.828063e+17         1
                    5.952785e+17         1
                    6.642713e+17         3
                    6.897633e+17         1
                    6.988815e+17        15
                    7.750695e+17         2
                    8.676168e+17         1
                    8.930982e+17         2
                    9.352790e+17         1
                    1.043926e+18         1
                    1.059358e+18         2
                    1.118515e+18         8
                    1.181318e+18        10
                    1.267677e+18        47
                    1.290332e+18         2
                    1.315852e+18         1
                    1.321681e+18         8
                    1.326366e+18         1
                    1.327864e+18         1
                    1.353619e+18         1
...                                    ...
6333597102633388268 8.370630e+18         3
                    8.400757e+18         4
                    8.421177e+18         2
                    8.504826e+18         2
                    8.528025e+18         2
                    8.531839e+18         1
                    8.535198e+18         1
                    8.557777e+18         2
                    8.604122e+18        18
                    8.645065e+18         3
                    8.656854e+18         1
                    8.668567e+18         2
                    8.716277e+18         8
                    8.733769e+18         9
                    8.830548e+18         1
                    8.845526e+18         1
                    8.891413e+18         2
                    8.908239e+18         1
                    9.053558e+18         1
                    9.081824e+18         1
                    9.096051e+18         1
                    9.098443e+18        36
                    9.109466e+18        17
                    9.149036e+18         4
                    9.158814e+18         1
                    9.177626e+18         3
                    9.179743e+18         1
                    9.217611e+18         1
                    9.221733e+18         6
                    9.222450e+18         2

[499 rows x 1 columns]

# Dispositivos por country

## Dispositivos country = 2970470518450881158

In [136]:
dispositivos_country1 = dispositivos.loc[dispositivos.device_countrycode == 2970470518450881158]
dispositivos_country1.head()

created   device_countrycode  device_brand  device_model  \
135  2019-03-10 00:31:15.940  2970470518450881158           NaN  5.613022e+18   
136  2019-03-10 23:18:21.729  2970470518450881158           NaN  7.628197e+18   
137  2019-03-11 22:41:12.621  2970470518450881158           NaN  7.787531e+18   
138  2019-03-13 11:58:55.848  2970470518450881158           NaN  2.066431e+17   
139  2019-03-13 00:47:01.874  2970470518450881158           NaN  2.810099e+17   

     device_language  Cantidad  
135     3.301378e+18         1  
136     3.301378e+18         1  
137     4.077062e+17         1  
138     3.301378e+18         1  
139     3.301378e+18         1

### Device_brand 

In [137]:
dispos = dispositivos_country1.fillna(value="VACIO")

In [138]:
coutry1_device_brand = dispos.groupby("device_brand").agg({'Cantidad': 'sum'}).reset_index()
coutry1_device_brand.sort_values(ascending=False,by="Cantidad").head(10)

device_brand  Cantidad
0        VACIO      1592

In [139]:
#Todos los datos de device_brand dentro de este pais estan vacios

### Device_model

In [140]:
country1_device_model = dispositivos_country1.groupby("device_model").agg({'Cantidad': 'count'}).reset_index()
country1_device_model.sort_values(ascending=False,by="Cantidad").head(10)

device_model  Cantidad
172  6.882415e+18       273
67   2.616228e+18        70
72   2.928262e+18        57
231  9.149036e+18        52
1    1.187661e+17        48
23   1.267677e+18        47
55   2.136935e+18        38
224  8.656854e+18        35
148  5.990117e+18        32
76   2.983638e+18        31

In [141]:
#Cantidad de modelos
country1_device_model.device_model.count()

234

### Device Lenguage

In [142]:
country1_device_lenguage = dispositivos_country1.groupby("device_language").agg({'Cantidad': 'sum'}).reset_index()
country1_device_lenguage.sort_values(ascending=False,by="Cantidad").head(10)

device_language  Cantidad
2     3.301378e+18      1475
0     4.077062e+17        69
3     4.060930e+18        47
1     5.794427e+17         1

In [143]:
#Cantidad de idiomas
country1_device_lenguage.device_language.count()

4

## Dispositivos country = 6333597102633388268

In [144]:
dispositivos_country2 = dispositivos.loc[dispositivos.device_countrycode == 6333597102633388268]
dispositivos_country2.head()

created   device_countrycode  device_brand  device_model  \
0  2019-03-13 01:43:33.445  6333597102633388268  3.083059e+17  4.445252e+18   
1  2019-03-13 04:58:35.078  6333597102633388268  3.083059e+17  4.445252e+18   
2  2019-03-13 04:20:57.666  6333597102633388268  5.137992e+17  6.026543e+18   
3  2019-03-13 04:20:57.698  6333597102633388268  5.137992e+17  6.026543e+18   
4  2019-03-10 22:24:56.684  6333597102633388268  1.083369e+18  7.876612e+18   

   device_language  Cantidad  
0              NaN         1  
1              NaN         1  
2     3.301378e+18         1  
3              NaN         1  
4     3.301378e+18         1

### Device Brand

In [145]:
country2_device_brand = dispositivos_country2.groupby("device_brand").agg({'Cantidad': 'sum'}).reset_index()
country2_device_brand.sort_values(ascending=False,by="Cantidad").head(10)

device_brand  Cantidad
0   3.083059e+17       472
4   2.208835e+18       185
6   2.523246e+18       116
21  5.951325e+18        72
8   2.987569e+18        40
14  3.812621e+18        37
1   5.137992e+17        34
23  6.538562e+18        26
11  3.228516e+18         9
3   1.083369e+18         9

In [146]:
#Hay 27 marcas
country2_device_brand.device_brand.count()

27

### Device Model

In [159]:
country2_device_model = dispositivos_country2.groupby("device_model").agg({'Cantidad': 'sum'}).reset_index()
country2_device_model.sort_values(ascending=False,by="Cantidad").head(10)

device_model  Cantidad
60   2.331947e+18       759
221  7.787531e+18        37
203  7.116343e+18        36
256  9.098443e+18        36
3    2.066431e+17        31
39   1.754528e+18        22
190  6.642035e+18        19
243  8.604122e+18        18
134  4.801467e+18        17
124  4.430579e+18        17

In [160]:
#Cantidad de modelos
country2_device_model.device_model.count()

265

### Device_language

In [149]:
country2_device_language = dispositivos_country2.groupby("device_language").agg({'Cantidad': 'sum'}).reset_index()
country2_device_language.sort_values(ascending=False,by="Cantidad").head(10)

device_language  Cantidad
28     9.028383e+18       714
17     4.621024e+18       632
25     6.977049e+18       238
5      1.526421e+18        56
1      2.822843e+17        39
20     6.035180e+18        24
4      1.193279e+18        21
21     6.535228e+18        10
12     3.301378e+18         9
0      1.101539e+17         9

In [150]:
#Cantidad de lenguajes
country2_device_language.device_language.count()

29

# JOINS

In [175]:
join_countrys = country2_device_model.merge(country1_device_model , how="left",on="device_model").fillna(0)

In [184]:
join_countrys.columns = ["device_model","6333597102633388268", "2970470518450881158"]
join_countrys.sort_values(ascending=False,by="2970470518450881158").head(10)

device_model  6333597102633388268  2970470518450881158
196  6.882415e+18                    3                273.0
71   2.616228e+18                    6                 70.0
78   2.928262e+18                    6                 57.0
258  9.149036e+18                    4                 52.0
2    1.187661e+17                    7                 48.0
25   1.267677e+18                    6                 47.0
53   2.136935e+18                    3                 38.0
245  8.656854e+18                    1                 35.0
174  5.990117e+18                   14                 32.0
82   2.983638e+18                    7                 31.0

In [185]:
join_countrys.sort_values(ascending=False,by="6333597102633388268").head(10)

device_model  6333597102633388268  2970470518450881158
60   2.331947e+18                  759                  0.0
221  7.787531e+18                   37                 28.0
203  7.116343e+18                   36                  4.0
256  9.098443e+18                   36                  0.0
3    2.066431e+17                   31                  9.0
39   1.754528e+18                   22                 16.0
190  6.642035e+18                   19                  0.0
243  8.604122e+18                   18                  1.0
134  4.801467e+18                   17                  2.0
124  4.430579e+18                   17                  1.0

No hay consistencia entre los dos paises en cuanto a los installs de los dispositivos con mayores cantidades.

Para el device "2.331947e+18" hay 759 installs en el pais 6333597102633388268 pero para el otro pais hay cantidad igual a 0. Se ve que algunos dispositivos no existen en el otro pais.

In [188]:
#Veo cuantos devide_model hay en cada pais y cuantos lenguajes por cada dispositivo
dispositivos.groupby(["device_countrycode","device_model","device_language"]).agg({'Cantidad': 'sum'})

Cantidad
device_countrycode  device_model device_language          
2970470518450881158 1.065211e+17 3.301378e+18            1
                    1.187661e+17 4.077062e+17            3
                                 3.301378e+18           42
                                 4.060930e+18            3
                    1.819881e+17 3.301378e+18            2
                    2.066431e+17 3.301378e+18            9
                    2.365619e+17 3.301378e+18            1
                    2.642581e+17 4.077062e+17            1
                                 3.301378e+18           12
                                 4.060930e+18            2
                    2.810099e+17 4.077062e+17            1
                                 3.301378e+18           11
                    3.403532e+17 3.301378e+18           21
                    3.686067e+17 3.301378e+18            4
                    3.817244e+17 3.301378e+18            1
                    3.828063e+17 3.301378e+18            1
                    5.952785e+17 3.301378e+18            1
                    6.642713e+17 3.301378e+18            2
                                 4.060930e+18            1
                    6.897633e+17 3.301378e+18            1
                    6.988815e+17 3.301378e+18           15
                    7.750695e+17 4.077062e+17            1
                                 3.301378e+18            1
                    8.676168e+17 3.301378e+18            1
                    8.930982e+17 3.301378e+18            2
                    9.352790e+17 3.301378e+18            1
                    1.043926e+18 3.301378e+18            1
                    1.059358e+18 3.301378e+18            1
                                 4.060930e+18            1
                    1.118515e+18 3.301378e+18            8
...                                                    ...
6333597102633388268 8.531839e+18 9.028383e+18            1
                    8.535198e+18 6.977049e+18            1
                    8.557777e+18 1.526421e+18            1
                    8.604122e+18 9.028383e+18           18
                    8.645065e+18 6.977049e+18            3
                    8.656854e+18 9.028383e+18            1
                    8.668567e+18 6.977049e+18            1
                                 9.028383e+18            1
                    8.716277e+18 6.977049e+18            2
                                 9.028383e+18            6
                    8.733769e+18 6.977049e+18            9
                    8.830548e+18 9.028383e+18            1
                    8.845526e+18 6.977049e+18            1
                    8.891413e+18 9.028383e+18            2
                    8.908239e+18 9.028383e+18            1
                    9.053558e+18 9.028383e+18            1
                    9.081824e+18 9.028383e+18            1
                    9.096051e+18 9.028383e+18            1
                    9.098443e+18 1.686906e+18            1
                                 9.028383e+18           35
                    9.109466e+18 9.028383e+18           17
                    9.149036e+18 1.526421e+18            1
                                 4.621024e+18            2
                    9.158814e+18 9.028383e+18            1
                    9.177626e+18 9.028383e+18            3
                    9.179743e+18 4.073835e+18            1
                    9.217611e+18 9.028383e+18            1
                    9.221733e+18 6.977049e+18            3
                                 9.028383e+18            3
                    9.222450e+18 9.028383e+18            2

[637 rows x 1 columns]

In [15]:
#Dispositivos por pais
dispositivos.groupby(["device_model","device_countrycode"]).agg({'Cantidad': 'sum'})

Cantidad
device_model device_countrycode           
5.112505e+16 6333597102633388268         9
7.252441e+16 6333597102633388268         1
1.065211e+17 2970470518450881158         1
1.187661e+17 2970470518450881158        48
             6333597102633388268         7
1.819881e+17 2970470518450881158         2
2.066431e+17 2970470518450881158         9
             6333597102633388268        31
2.091433e+17 6333597102633388268         2
2.365619e+17 2970470518450881158         1
2.642581e+17 2970470518450881158        15
             6333597102633388268         3
2.810099e+17 2970470518450881158        12
             6333597102633388268         5
3.037516e+17 6333597102633388268         3
3.403532e+17 2970470518450881158        21
3.686067e+17 2970470518450881158         4
             6333597102633388268         7
3.817244e+17 2970470518450881158         1
3.828063e+17 2970470518450881158         1
4.547631e+17 6333597102633388268         4
5.491128e+17 6333597102633388268         1
5.952785e+17 2970470518450881158         1
6.025259e+17 6333597102633388268         3
6.561250e+17 6333597102633388268         2
6.642713e+17 2970470518450881158         3
6.897633e+17 2970470518450881158         1
6.953092e+17 6333597102633388268         3
6.988815e+17 2970470518450881158        15
7.286156e+17 6333597102633388268         1
...                                    ...
8.656854e+18 2970470518450881158        35
             6333597102633388268         1
8.668567e+18 6333597102633388268         2
8.699294e+18 2970470518450881158         6
8.716277e+18 6333597102633388268         8
8.733769e+18 6333597102633388268         9
8.830548e+18 6333597102633388268         1
8.845526e+18 2970470518450881158         1
             6333597102633388268         1
8.877195e+18 2970470518450881158         1
8.891413e+18 6333597102633388268         2
8.908239e+18 6333597102633388268         1
9.053558e+18 6333597102633388268         1
9.074460e+18 2970470518450881158         1
9.081824e+18 6333597102633388268         1
9.092894e+18 2970470518450881158         2
9.096051e+18 6333597102633388268         1
9.098097e+18 2970470518450881158         1
9.098443e+18 6333597102633388268        36
9.109466e+18 6333597102633388268        17
9.149036e+18 2970470518450881158        52
             6333597102633388268         4
9.158814e+18 6333597102633388268         1
9.177626e+18 6333597102633388268         3
9.179743e+18 2970470518450881158         1
             6333597102633388268         1
9.217611e+18 6333597102633388268         1
9.221733e+18 2970470518450881158         5
             6333597102633388268         6
9.222450e+18 6333597102633388268         2

[499 rows x 1 columns]

In [16]:
#Ver cada dispositivo en que pais se encuentra y cuantos lenguajes tiene
dispositivos.groupby(["device_model","device_countrycode","device_language"]).agg({'Cantidad': 'sum'})

Cantidad
device_model device_countrycode  device_language          
5.112505e+16 6333597102633388268 6.977049e+18            2
                                 9.028383e+18            7
7.252441e+16 6333597102633388268 6.977049e+18            1
1.065211e+17 2970470518450881158 3.301378e+18            1
1.187661e+17 2970470518450881158 4.077062e+17            3
                                 3.301378e+18           42
                                 4.060930e+18            3
             6333597102633388268 1.526421e+18            2
                                 4.621024e+18            4
1.819881e+17 2970470518450881158 3.301378e+18            2
2.066431e+17 2970470518450881158 3.301378e+18            9
             6333597102633388268 6.977049e+18            4
                                 9.028383e+18           27
2.091433e+17 6333597102633388268 6.977049e+18            2
2.365619e+17 2970470518450881158 3.301378e+18            1
2.642581e+17 2970470518450881158 4.077062e+17            1
                                 3.301378e+18           12
                                 4.060930e+18            2
             6333597102633388268 4.621024e+18            2
2.810099e+17 2970470518450881158 4.077062e+17            1
                                 3.301378e+18           11
             6333597102633388268 6.977049e+18            4
                                 9.028383e+18            1
3.037516e+17 6333597102633388268 9.028383e+18            3
3.403532e+17 2970470518450881158 3.301378e+18           21
3.686067e+17 2970470518450881158 3.301378e+18            4
             6333597102633388268 6.977049e+18            3
                                 9.028383e+18            4
3.817244e+17 2970470518450881158 3.301378e+18            1
3.828063e+17 2970470518450881158 3.301378e+18            1
...                                                    ...
8.733769e+18 6333597102633388268 6.977049e+18            9
8.830548e+18 6333597102633388268 9.028383e+18            1
8.845526e+18 2970470518450881158 3.301378e+18            1
             6333597102633388268 6.977049e+18            1
8.877195e+18 2970470518450881158 3.301378e+18            1
8.891413e+18 6333597102633388268 9.028383e+18            2
8.908239e+18 6333597102633388268 9.028383e+18            1
9.053558e+18 6333597102633388268 9.028383e+18            1
9.074460e+18 2970470518450881158 3.301378e+18            1
9.081824e+18 6333597102633388268 9.028383e+18            1
9.092894e+18 2970470518450881158 3.301378e+18            2
9.096051e+18 6333597102633388268 9.028383e+18            1
9.098097e+18 2970470518450881158 4.077062e+17            1
9.098443e+18 6333597102633388268 1.686906e+18            1
                                 9.028383e+18           35
9.109466e+18 6333597102633388268 9.028383e+18           17
9.149036e+18 2970470518450881158 4.077062e+17            2
                                 3.301378e+18           49
                                 4.060930e+18            1
             6333597102633388268 1.526421e+18            1
                                 4.621024e+18            2
9.158814e+18 6333597102633388268 9.028383e+18            1
9.177626e+18 6333597102633388268 9.028383e+18            3
9.179743e+18 2970470518450881158 3.301378e+18            1
             6333597102633388268 4.073835e+18            1
9.217611e+18 6333597102633388268 9.028383e+18            1
9.221733e+18 2970470518450881158 3.301378e+18            5
             6333597102633388268 6.977049e+18            3
                                 9.028383e+18            3
9.222450e+18 6333597102633388268 9.028383e+18            2

[637 rows x 1 columns]

# Auctions

In [30]:
auctions = pd.read_csv('./Datos2019/auctions.csv')

auctions.head()

auction_type_id              country                        date  \
0              NaN  6333597102633388268  2019-03-11 14:18:33.290763   
1              NaN  6333597102633388268  2019-03-11 14:18:34.440157   
2              NaN  6333597102633388268  2019-03-11 14:18:35.862360   
3              NaN  6333597102633388268  2019-03-11 14:18:36.167163   
4              NaN  6333597102633388268  2019-03-11 14:18:37.728590   

             device_id  platform  ref_type_id  source_id  
0  6059599345986491085         1            1          0  
1  1115994996230693426         1            1          0  
2  7463856250762200923         1            1          0  
3  7829815350267792040         1            1          0  
4  1448534231953777480         1            1          0

In [ ]:
auctions.country.value_counts()